### PR TITLE
Remove rollup-typescript

### PR DESCRIPTION
**Pull request by matklad**
It seems like just calling typescript directly is simpler and more reliable?

@Veetaha what do you think about this approach?

### CODE REVIEW

**Comment by Veetaha** [comment]
I agree, but you should also remove `@rollup/plugin-typescript` from `rollup.config.js`.

File: editors/code/package.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
         "vscode": "^1.41.0"
     },
     "scripts": {
-        "vscode:prepublish": "rollup -c",
+        "vscode:prepublish": "tsc -p ./ && rollup -c",
```
Comment (Veetaha) [inline editors/code/package.json]: there is no need for `-p ./` parameter here, just `tsc` will work

**Comment by matklad** [comment]
It is removed: https://github.com/rust-analyzer/rust-analyzer/pull/3003/files#diff-b08397d7bf1bf80738ae36841379c7a5L1

Or is there some other place where I could remove it from? I'd be glad, I like removing things :)

**Comment by matklad** [comment]
>I'd also change the output file of rollup so that it doesn't overwrite the existing out/main.js file

That's kind of a deliberate hack: we have to name output file in `package.json`, and I want the extension to work with and without bundling.

**Comment by Veetaha** [comment]
> That's kind of a deliberate hack: we have to name output file in package.json, and I want the extension to work with and without bundling.

Yes, good point, this is the detail I've missed

**Comment by Veetaha** [comment]
However, LGTM, unless it doesn't compile or doesn't bundle ;)

**Comment by matklad** [comment]
bors r+

It seems to work for both of these cases

**Comment by Veetaha** [comment]
Or doesn't work at runtime, or doesn't run when bundled ...

**Comment by matklad** [comment]
We might try to re-introduce the tslib in a proper way now...

**Pull request merged**